### PR TITLE
Runtime: Core BPF: Add test for CPI post-migration

### PR DESCRIPTION
**Pull request by buffalojoec**
#### Problem
Our test suite for migrating builtin programs to BPF sends a transaction with an instruction for the newly migrated program, to test that it can be loaded and executed successfully. However, this test suite is missing a test for CPI'ing to this program.

Building on the back of #2483, we should have coverage for this case.

#### Summary of Changes
Add a mocked-out builtin processor designed to CPI to the provided program to the test suite. Then simply send a transaction to this builtin (like we're already doing with the newly migrated program) directing it to CPI to the newly migrated program.

### CODE REVIEW

**Review by CriesofCarrots — APPROVED** [review]
Lgtm. I'd get @2501babe 's sign-off as well, though

**Comment by 2501babe** [comment]
does this new test fail if you turn off your fix (ie if the builtin is not marked executable)? the check im aware of that would fail this is before control drops to invoke context, though there might be another check inside it

not that you shouldnt merge this, i think a lot more testing of transaction loading needs to take place and im brainstorming myself how to do that

**Review by 2501babe — APPROVED** [review]

**Comment by buffalojoec** [comment]
> does this new test fail if you turn off your fix (ie if the builtin is not marked executable)? the check im aware of that would fail this is before control drops to invoke context, though there might be another check inside it

Yep. Directly invoking it works, but via CPI it throws `InstructionError(0, AccountNotExecutable)`.

Looks to me like this callsite:

https://github.com/anza-xyz/agave/blob/ae18213c19ea5335dfc75e6b6116def0f0910aff/program-runtime/src/invoke_context.rs#L437-L439

**Pull request merged**
